lib/m2ts/caption-stream.js

```diff
@@ -491,7 +492,14 @@ var Cea608Stream = function(field, dataChannel) {
       this.nonDisplayed_ = createDisplayBuffer();
 
     } else if (data === this.RESUME_DIRECT_CAPTIONING_) {
+      if (this.mode_ !== 'paintOn') {
+        // NOTE: This should be removed when proper caption positioning is
+        // implemented
+        this.flushDisplayed(packet.pts);
+        this.displayed_ = createDisplayBuffer();
```

Comment:
It is non-compliant with the spec, but unfortunately we can't be compliant because of the way formatting is currently being done (replacing PACs/mid-row codes with HTML codes on reception). It's best to clear it out IMO -- that way you'll never end up with something like `caption <imore text`. I figure it's more likely to want your paint-on caption to disappear over having it stick, but, /shrug.

Reply:
You mean have your last non-paint-on caption disappear, right?

Reply:
Yes, my mistake, that is what I meant.